src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagEntryValidator.cs

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Linq;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.Query;
+
+namespace Microsoft.Health.Dicom.Core.Features.CustomTag
+{
+    public class CustomTagEntryValidator : ICustomTagEntryValidator
+    {
+        private const string UnknownTagName = "Unknown";
+
+        /* Unsupported VRCodes:
+         LT(Long Text), OB (Other Byte), OD (Other Double), OF(Other Float), OL (Other Long), OV(other Very long), OW (other Word), ST(Short Text, SV (Signed Very long)
+         UC (Unlimited Characters), UN (Unknown), UR (URI), UT (Unlimited Text), UV (Unsigned Very long)
+         Note: we dont' find definition for UR, UV and SV in DICOM standard (http://dicom.nema.org/dicom/2013/output/chtml/part05/sect_6.2.html)
+        */
+
+        public static readonly IImmutableSet<string> SupportedVRCodes = ImmutableHashSet.Create(
+            DicomVRCode.AE,
+            DicomVRCode.AS,
+            DicomVRCode.AT,
+            DicomVRCode.CS,
+            DicomVRCode.DA,
+            DicomVRCode.DS,
+            DicomVRCode.DT,
+            DicomVRCode.FD,
+            DicomVRCode.FL,
+            DicomVRCode.IS,
+            DicomVRCode.LO,
+            DicomVRCode.PN,
+            DicomVRCode.SH,
+            DicomVRCode.SL,
+            DicomVRCode.SS,
+            DicomVRCode.TM,
+            DicomVRCode.UI,
+            DicomVRCode.UL,
+            DicomVRCode.US);
+
+        public void ValidateCustomTags(IEnumerable<CustomTagEntry> customTagEntries)
+        {
+            EnsureArg.IsNotNull(customTagEntries, nameof(customTagEntries));
+            if (customTagEntries.Count() == 0)
+            {
+                throw new CustomTagEntryValidationException(DicomCoreResource.MissingCustomTag);
+            }
+
+            HashSet<string> pathSet = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            foreach (CustomTagEntry tagEntry in customTagEntries)
+            {
+                ValidateCustomTagEntry(tagEntry);
+
+                // don't allow duplicated path
+                if (pathSet.Contains(tagEntry.Path))
+                {
+                    throw new CustomTagEntryValidationException(
+                         string.Format(CultureInfo.InvariantCulture, DicomCoreResource.DuplicateCustomTag, tagEntry.Path));
+                }
+
+                pathSet.Add(tagEntry.Path);
+            }
+        }
+
+        private void ValidateCustomTagEntry(CustomTagEntry tagEntry)
+        {
```

Comment:
Can you add function summaries on these?

Reply:
I'm not sure what methods "these" refer to.   I think as last time team discussion, only class level summary are mandatory.
I will add summary on this method

Reply:
I think it's mostly since there are many functions here that do similar sounding things!

Reply:
I see. thanks Sneha. :)